File: ueueo-framework/ueueo-localization/src/main/java/Abp/Localization/LanguageInfoExtensions.cs
using System.Collections.Generic;
using System.Linq;
using JetBrains.Annotations;

namespace Volo.Abp.Localization;

public static class LanguageInfoExtensions
{
    public static T FindByCulture<T>(
        @Nonnull this IEnumerable<T> languages,
        @Nonnull String cultureName,
        @Nullable String uiCultureName = null)
    where T : class, ILanguageInfo
    {
        if (uiCultureName == null)
        {
            uiCultureName = cultureName;
        }

        var languageList = languages.ToList();

        return languageList.FirstOrDefault(l => l.CultureName == cultureName && l.UiCultureName == uiCultureName)
               ?? languageList.FirstOrDefault(l => l.CultureName == cultureName)
               ?? languageList.FirstOrDefault(l => l.UiCultureName == uiCultureName);
    }
}

File: ueueo-framework/ueueo-localization/src/main/java/com/ueueo/localization/LanguageInfoExtensions.java
package com.ueueo.localization;

import org.apache.commons.lang3.StringUtils;
import org.springframework.lang.NonNull;
import org.springframework.lang.Nullable;

import java.util.List;

public class LanguageInfoExtensions {

    public static <T extends ILanguageInfo> T findByCulture(
            @NonNull List<T> languages,
            @NonNull String cultureName,
            @Nullable String uiCultureName) {
        String uiCulture = StringUtils.isNoneBlank(uiCultureName) ? uiCultureName : cultureName;

        T language = languages.stream().filter(l -> l.getCultureName().equals(cultureName)
                        && l.getUiCultureName().equals(uiCulture))
                .findFirst()
                .orElse(null);
        if (language == null) {
            language = languages.stream().filter(l -> l.getCultureName().equals(cultureName))
                    .findFirst()
                    .orElse(null);
        }
        if (language == null) {
            language = languages.stream().filter(l -> l.getUiCultureName().equals(uiCulture))
                    .findFirst()
                    .orElse(null);
        }
        return language;
    }
}
